flink-core/src/main/java/org/apache/flink/types/parser/BigDecParser.java

```diff
@@ -55,9 +56,20 @@ public int parseField(byte[] bytes, int startPos, int limit, byte[] delimiter, B
 			return -1;
 		}
 
-		String str = new String(bytes, startPos, i - startPos);
 		try {
-			this.result = new BigDecimal(str);
+			final int length = i - startPos;
+			if (reuse == null || reuse.length != length) {
```

Comment:
can we change this to `reuse.length < length` to only create a new array in case more space is needed?